packages/wallet/test/wallet.spec.ts

```diff
@@ -155,6 +156,37 @@ describe('Wallet', () => {
         expect(tx).toBeDefined();
     });
 
+    it('should match UTXOs from a silent block without relying on db', async () => {
+        const scanKey = wallet['masterKey'].derivePath("m/352'/1'/0'/1'/0");
+        const spendKey = wallet['masterKey'].derivePath("m/352'/1'/0'/0'/0");
+
+        if (!scanKey.privateKey)
+            throw new Error('Scan private key is undefined');
+
+        const matchedUTXOs = wallet.matchSilentBlockOutputs(
+            parsedSilentBlock,
+            scanKey.privateKey,
+            scanKey.publicKey,
+            spendKey.publicKey,
+        );
+        expect(matchedUTXOs.length).toBeGreaterThan(0);
+        expect(matchedUTXOs[0]).toHaveProperty('txid');
+    });
+
+    it('should scan a silent block and update UTXOs', async () => {
```

Comment:
we should add more tests in the future, please use the standard `it.each` pattern here